components/remote_settings/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+pub mod error;
+use config::RemoteSettingsConfig;
+pub use error::{RemoteSettingsError, Result};
+mod config;
+use rs_client::{Client, ClientConfig};
+use std::{fs::File, io::prelude::Write, sync::Mutex};
+
+uniffi_macros::include_scaffolding!("remotesettings");
+
+pub struct RemoteSettings {
+    pub config: RemoteSettingsConfig,
+    client: Mutex<Client>,
+}
+
+impl RemoteSettings {
+    pub fn new(config: RemoteSettingsConfig) -> Result<Self> {
+        Ok(RemoteSettings {
+            config: config.clone(),
+            client: Mutex::new(Client::new(config.into())?),
+        })
+    }
+
+    pub fn get_records(&self) -> Result<Vec<JsonObject>> {
+        let client = self.client.lock().unwrap();
+        let resp = client.get_records()?;
+        self.parse_records(resp.text().to_string())
+    }
+
+    pub fn download_attachment_to_path(
+        &self,
+        attachment_location: String,
+        path: String,
+    ) -> Result<()> {
+        let client = self.client.lock().unwrap();
+        let resp = client.get_attachment(&attachment_location)?;
+        let mut file = File::create(path)?;
```

Comment:
Is there something that is known to work? @mhammond and I talked about this a bit, but it seemed riskier to me to pass something like `Vec<u8>` across the FFI layer, since attachments can technically be of arbitrary size.

Reply:
This is probably fine, we definitely know we can open already existing files (we do that for our DBs for example) - so if we can ensure that the consumer has already created the file, writing to it should work!

Reply:
We create our DBs too. I thought the important thing was that they tell us the location?

Reply:
Eeeh nvm mind me, I'm confusing the opening/creating apis with some more specific file metadata apis that I've seen not work properly in some platforms